### PR TITLE
ProcessOrderbook: added a condition to check for existing orderbooks

### PR DESCRIPTION
**Pull request by andreygrehov**

### CODE REVIEW

File: exchanges/orderbook/orderbook.go [hunk 1]
```diff
@@ -178,13 +178,16 @@ func ProcessOrderbook(exchangeName string, p pair.CurrencyPair, orderbookNew Bas
 	}
 
 	if FirstCurrencyExists(exchangeName, p.FirstCurrency) {
-		if !SecondCurrencyExists(exchangeName, p) {
+		if SecondCurrencyExists(exchangeName, p) {
+			m.Lock()
+			orderbook.Orderbook[p.FirstCurrency][p.SecondCurrency][orderbookType] = orderbookNew
+			m.Unlock()
+			return
+		} else {
```
[the remark this answers is not in the document]
Comment (andreygrehov) [reply exchanges/orderbook/orderbook.go]: see 48b143a

File: exchanges/orderbook/orderbook_test.go [hunk 2]
```diff
@@ -263,4 +266,62 @@ func TestProcessOrderbook(t *testing.T) {
...
+			bids := []Item{{Price: rand.Float64(), Amount: rand.Float64()}}
+			base := Base{
+				Pair:         newPairs,
+				CurrencyPair: newPairs.Pair().String(),
+				Asks:         asks,
+				Bids:         bids,
+			}
+
+			ProcessOrderbook(newName, newPairs, base, Spot)
+			testArray = append(testArray, quick{Name: newName, P: newPairs, Bids: bids, Asks: asks})
```
Comment (shazbert) [inline exchanges/orderbook/orderbook_test.go]: Races should only occur on the track :P.  Could you please add a mutex around this append?

**Comment by shazbert** [comment]
@andreygrehov thrasher is going to push another update on to the test and then this will be merged.

**Comment by andreygrehov** [comment]
yea, got a conflict already :) Thank you.

**Pull request merged**